diesel_tests/tests/filter_operators.rs

```diff
@@ -264,6 +264,36 @@ fn filter_by_in() {
     );
 }
 
+#[test]
+#[cfg(feature = "postgres")]
+fn filter_array_by_in() {
+    use crate::schema::posts::dsl::*;
+
+    let connection: &mut PgConnection = &mut connection();
+    let tag_combinations_to_look_for: &[&[&str]] = &[&["foo"], &["foo", "bar"], &["baz"]];
+    let result: Vec<i32> = posts
+        .filter(tags.eq_any(tag_combinations_to_look_for))
```

Comment:
we probably want to document this behavior in these places:
https://github.com/diesel-rs/diesel/blob/5a2940eeaa935c0cc0a7fc57ffe012f3aaa80f43/diesel/src/expression/array_comparison.rs#L31-L32
https://github.com/diesel-rs/diesel/blob/5a2940eeaa935c0cc0a7fc57ffe012f3aaa80f43/diesel/src/expression/array_comparison.rs#L51-L52
https://github.com/diesel-rs/diesel/blob/5a2940eeaa935c0cc0a7fc57ffe012f3aaa80f43/diesel/src/expression_methods/global_expression_methods.rs#L109-L110